CONTRIBUTING.md

```diff
@@ -63,6 +63,44 @@ quickly merge or address your contributions.
 
 ## Pull requests
 
+First, consider opening an issue (see above) to discuss the approach you'd like 
```

Comment:
Should we put more emphasis on this?  It's always unfortunate when someone invests a lot of time on a PR that we won't accept.

Reply:
I like the way they put it in the Vault UI section: not "consider" but "please do this".

Reply:
Okay, will change.